lib/AST/Type.cpp

```diff
@@ -4046,21 +4047,31 @@ bool UnownedStorageType::isLoadable(ResilienceExpansion resilience) const {
   return ty->usesNativeReferenceCounting(resilience);
 }
 
-static bool doesOpaqueClassUseNativeReferenceCounting(const ASTContext &ctx) {
-  return !ctx.LangOpts.EnableObjCInterop;
-}
-
-static bool usesNativeReferenceCounting(ClassDecl *theClass,
-                                        ResilienceExpansion resilience) {
+static ReferenceCounting getClassReferenceCounting(
+                                             ClassDecl *theClass,
+                                             ResilienceExpansion resilience) {
   // TODO: Resilience? there might be some legal avenue of changing this.
   while (auto superclass = theClass->getSuperclassDecl()) {
     theClass = superclass;
   }
-  return !theClass->hasClangNode();
+
+  return theClass->hasClangNode()
+           ? ReferenceCounting::ObjC
+           : ReferenceCounting::Native;
 }
 
-bool TypeBase::usesNativeReferenceCounting(ResilienceExpansion resilience) {
+ReferenceCounting TypeBase::getReferenceCounting(
+                                              ResilienceExpansion resilience) {
```

Comment:
I don't think @jrose-apple wants to allow changing the superclass of a class from Objective-C to non-Objective-C... wouldn't it mess up too many things? Might want to just drop the resilience expansion parameter here

Reply:
Arguably, "insert superclasses in the hierarchy" could include "insert a new base class", but I don't think that's a *good* idea or anything.

Reply:
I'll kill off this parameter. We're not using it now, and it doesn't seem that we're likely to do so.